languages/csharp/exercises/concept/tuples/.docs/introduction.md

```diff
@@ -0,0 +1,69 @@
+In C#, a tuple is a data structure which organizes data, holding two or more fields
+of any type.
+
+A tuple is typically created by placing 2 or more expressions separated by comas,
+within a set of parentheses.
+
+```csharp
+string boast = "All you need to know";
+bool success = !string.IsNullOrWhiteSpace(boast);
+(bool, int, string) tripple = (success, 42, boast);
+```
+
+A tuple can be used in assignment and intialization operations, as a return value or a method
+argument and in equality testing.
+
+Fields are extracted using dot syntax. By default, the first field is `Item1`,
+the second `Item2`, etc. Non-default names are discussed below.
+
+```csharp
+// initialization
+
+(int, int, int) vertices = (90, 45, 45);
+
+// assignment
+
+vertices = (60, 60, 60);
+
+//  return value
+
+(bool, int) GetSameOrBigger(int num1, int num2)
+{
+    return (num1 == num2, num1 > num2 ? num1 : num2);
+}
+
+// method argument
+
+int Add((int, int) operands)
+{
+    return operands.Item1 + operands.Item2;
+}
+
+
+// equality testing
+
+var estimateA = (42, 1729);
+var estimateB = (2*3*7, 7*13*19);
+bool result = estimateA == estimateB;
+// => result == true
+
+```
+
+Field names `Item1` etc. do not make for readable code. The code below shows
+2 ways to name the fields of tuples.
+
+```csharp
+// name items in declaration
+
+(bool success, string message) results = (true, "well done!");
+
+bool mySuccess = results.success;
+string myMessaage = results.message;
+
+// name items in creating expression
+
+var results2 = (success: true, message: "well done!");
```

Comment:
Maybe it would be worth mentioning explicitly somewhere in this document that the tuple type can be inferred automatically using `var`.

Reply:
Done